### PR TITLE
Fix warning for boolean props

### LINKED ISSUE #1628: Vue warning for boolean variables
Error message in the build console:
>---> <NestedPanel> at vue-components/src/panels/NestedPanel.vue
       <Panel> at vue-components/src/Panel.vue
         <Root>
[Vue warn]: Invalid prop: type check failed for prop "expandable". Expected Boolean, got String with value "true".
found in
---> <NestedPanel> at vue-components/src/panels/NestedPanel.vue
       <Panel> at vue-components/src/Panel.vue
         <Root>

To reproduce, 

1. clone https://github.com/nus-cs2103-AY2021S2/website
1. clone the submodule, as explained in the README
1. checkout the branch `cs2103`
1. run `markbind build`

I guess similar to #1610 

Remember to test against the cs2103 website in case there are similar errors but not exactly same as the one given above.

### PR DESCRIPTION
**Pull request by jonahtanjz**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
Resolves #1628.

Add `String` type to Boolean props with default value of `true`. This is because HTML attribute value is passed as `String`.
In this case, `false` will be rendered as `String`:
```.html
<panel header="..." expandable="false">
  ...
</panel>
```
In this case, `minimized` will be passed as `Boolean`:
```.html
<panel header="..." minimized>
  ...
</panel>
```

**Anything you'd like to highlight / discuss:**

Should we add `String` type to all Boolean props that are from HTML attributes? Currently I only see the need for props with default value of `true` as the only way to set it to `false` as a HTML attribute is to pass a `"false"` value. 
However, if a user input this,
```.html
<panel header="..." minimized="true">
  ...
</panel>
```
minimized will be rendered as `String`. Not sure if the above example is the correct usage of HTML in general. Should we leave this to the user such that if they want an attribute to be true, just include the attribute there without any value? The above example of passing a `true` value will trigger warning unless we include `String` type for all `Boolean` props coming from HTML.

**Testing instructions:**
Follow steps in #1628.

**Proposed commit message: (wrap lines at 72 characters)**
Fix warning for Boolean props

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No blatantly unrelated changes
- [ ] Pinged someone for a review!

### CODE REVIEW

**Comment by ang-zeyu** [comment]
> Should we leave this to the user such that if they want an attribute to be true, just include the attribute there without any value?

Agree with this. Going forward we should (have been for while already though) stick with this for new component props. It should be simpler for both the user and implementation wise.

For the previous existing attributes using this behaviour, it is a breaking change, so we can slate that for v4.
In the meantime I think the docs need some updating on `true / false` versus _"just put the property name there"_ though =X

> Should we add String type to all Boolean props that are from HTML attributes? 

It should rightly be added for all props using the `toBoolean` function (a relic from vue v1->v2 migration), but I think its ok to fix-this-as-we-go to minimise possible side effects, the warning from vue is not critical in comparison. But if you're up for investigating each usage let's do that 😀

**Comment by jonahtanjz** [comment]
Thanks for the review @ang-zeyu

> In the meantime I think the docs need some updating on true / false versus "just put the property name there" though

Yup agree. Will open another issue and PR for this :)

>It should rightly be added for all props using the toBoolean function (a relic from vue v1->v2 migration), but I think its ok to fix-this-as-we-go to minimise possible side effects, the warning from vue is not critical in comparison. But if you're up for investigating each usage let's do that

Ok got it! Let's go with this 👍

**Comment by jonahtanjz** [comment]
@ang-zeyu I have updated the PR to include `String` to all relevant props using `toBoolean`.

File: packages/vue-components/src/TabGroup.vue [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { coerce } from './utils/utils';
 export default {
   props: {
     disabled: {
-      type: Boolean,
+      type: [Boolean, String],
```
Comment (ang-zeyu) [inline packages/vue-components/src/TabGroup.vue:16]: 👍

**Review by ang-zeyu — COMMENTED** [review]
Lgtm!

**Review by ang-zeyu — APPROVED** [review]

**Pull request merged**